src/stream-tcp-list.c

```diff
@@ -890,6 +895,37 @@ void StreamTcpPruneSession(Flow *f, uint8_t flags)
     SCReturn;
 }
 
+/**
+ * \brief Adds the following information to the TcpSegment from the current
+ *  packet being processed: time values, packet length, pcap_cnt, and the
+ *  header data of the packet. This information is added to the TcpSegment so
+ *  that it can be used in pcap capturing (log-pcap-stream) to dump the tcp
+ *  session at the beginning of the pcap capture.
+ * \param seg TcpSegment where information is being stored.
+ * \param p Packet being processed.
+ */
+static void StreamTcpSegmentAddPacketData(TcpSegment *seg, Packet *p)
```

Comment:
I'd like to see this done differently, w/o a alloc per segment. I suppose we could start with a size "x" and grow it as needed and just not free it. That would give the flexibility that we may need for large headers (e.g. IPv6 with ext hdrs), while not doing as many allocs.

I also wonder if we can have a more generic support than just ethernet. The space between `GET_PKT_DATA(p)` and `p->payload` should hold the headers, irrespective of the linktype.